### PR TITLE
feat: device emulation

### PR DESCRIPTION
**Pull request by kumaraditya303**
Add support for devices

### CODE REVIEW

**Comment by kumaraditya303** [comment]
@mxschmitt Ready for review !

**Comment by kumaraditya303** [comment]
@mxschmitt Any chance you can review this ?

**Comment by pavelfeldman** [comment]
Let's split this into mypy changes and the actual feature.

File: tests/test_playwright.py [hunk 1]
```diff
@@ -316,27 +316,6 @@ def test_one(self):
     result.assert_outcomes(passed=1)
 
 
-def test_device_emulation(testdir: pytest.Testdir) -> None:
```
Comment (pavelfeldman) [inline tests/test_playwright.py:319]: This is still a good test, call it `test_browser_context_args`

**Review by mxschmitt — APPROVED** [review]

File: tests/test_playwright.py [hunk 2]
```diff
@@ -316,7 +316,7 @@ def test_one(self):
...
+def test_browser_context_arg(testdir: pytest.Testdir) -> None:
```
Comment (mxschmitt) [inline tests/test_playwright.py:319]:
```suggestion
def test_browser_context_args(testdir: pytest.Testdir) -> None:
```

File: pytest_playwright/pytest_playwright.py [hunk 3]
```diff
@@ -203,6 +207,11 @@ def browser_channel(pytestconfig: Any) -> Optional[str]:
     return pytestconfig.getoption("--browser-channel")
 
 
+@pytest.fixture(scope="session")
+def device(pytestconfig: Any) -> Optional[str]:
```
Comment (mxschmitt) [inline pytest_playwright/pytest_playwright.py:211]: why to define the device as a fixture? We read now the option in both places and the one in the device fixture is probably unused.

**Review by mxschmitt — APPROVED** [review]

**Pull request merged**